pkg/aws/throttle/throttler_test.go

```diff
@@ -0,0 +1,199 @@
+package throttle
+
+import (
+	"context"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/time/rate"
+	"net/http"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+)
+
+func Test_throttler_InjectHandlers(t *testing.T) {
+	throttler := &throttler{}
+	handlers := request.Handlers{}
+	throttler.InjectHandlers(&handlers)
+	assert.Equal(t, 1, handlers.Sign.Len())
+}
+
+func Test_throttler_beforeSign(t *testing.T) {
```

Comment:
Can we describe what this is verifying either as a comment or more descriptive test cases? Clearly I had questions :P

Reply:
added more descriptions :D